### ISSUE TITLE
FeedbackChallenge : Websocket link for Blockfrost API 

### ISSUE THREAD

**Issue by stephen-rowan** [Quality-Assurance-DAO/F5-Developer-ecosystem-Proposal#51]
## FeedbackChallenge :  [Websocket link for Blockfrost API](https://cardano.ideascale.com/a/dtd/Websocket-link-for-Blockfrost-API/352757-48088)

In the spirit of Project Catalyst's #FeedbackChallenge we are assessing to what extent the Catalyst Funding Process and it's Proposals are **open source** as part of the proposal [Quality-Assurance-DAO](https://cardano.ideascale.com/a/dtd/Quality-Assurance-DAO/351678-48088). Your proposal has been selected because "open source" is mentioned in it's text or comments.

We would appreciate it if you could respond to the following questions (in the reply to my comments below) : 

1.  **Open** - Is your Organisation open-source ?

> Reading thought your assessment documentation, I did not find the specific criteria to respond to this question. But we feel we're an open source organization. Everything that is funded from the community is and will be open source. However, as we're a service provider, certain parts of our organization are internal, such as the infrastructure configuration or the backends. We have the plan to open source the backends and we have made a public promise to do so, we just do not have the capacity right now to do so.

2.  **Accessible** -  Can community contributors issue pull requests or raise issues on your source code ?

> Yes, our code is hosted on Github (not much yet, but it is slowly growing).

https://github.com/fivebinaries

https://github.com/blockfrost

3.  **Transparent** - Can community contributors access and propose changes to the source text of your documentation ?

> Yes, the specification of our backends is open source and the community's contributions are welcome (we already got one!) https://github.com/blockfrost/openapi.

To keep the assessment transparent each comment is recorded as a [GitHub Issue]( https://github.com/Quality-Assurance-DAO/F5-Developer-ecosystem-Proposal/issues/51 ) on **Quality-Assurance-DAO's** proposal repository (Apache-2.0 License) where a more detailed explanation can be found. 

Together we can build a community of open source contributors on the Cardano platform.

Thank You 

Stephen Whitenstall, QA-DAO

## Detailed explanation 

Your [Issue reference](https://github.com/Quality-Assurance-DAO/F5-Developer-ecosystem-Proposal/issues/50) includes a more detailed explanation of the methodology of this FeedbackChallenge : Open Source. There is also a *Open Source Assessment* [Project Board](https://github.com/Quality-Assurance-DAO/F5-Developer-ecosystem-Proposal/projects/3) and a GitHub bot post Issues in my channel #quality-assurance-dao on Project Catalyst's Discord.

We believe this is important because many developers are concerned with the lack of a clear commitment to open source and fear their free contributions will be locked into a proprietary platform.

## Open Source Assessment Method
- [x] Draft a basic Open Source Assessment Method.
https://github.com/Quality-Assurance-DAO/DAO-Open-Source/tree/main/Documents/Assessments

### Basic Open Source Assessment
1.  **Open** - Is your Organisation open-source ?
1.  **Accessible** -  Can your users issue pull requests or raise issues on your organisation's source code ?
1.  **Transparent** -  Can your users access and propose changes to the source of your organisation's documentation ?

### Method
- [x] Do text search of "open source" on Catalyst Projects.
- [x] Post assessment questions in comments of proposal.
- [x] Raise Issue for each proposal's assessment.
- [x] Record response to assessment.
- [x] Close Issue once proposal submitted and record responses

### Report
- [ ] Collate and publish report of findings with a Creative Commons License

**Comment by mmahut** [comment]
Thank you, this is a great initiative!

**1. Open - Is your Organisation open-source ?**

Reading thought your assessment documentation, I did not find the specific criteria to respond to this question. But we feel we're an open source organization. Everything that is funded from the community is and will be open source. However, as we're a service provider, certain parts of our organization are internal, such as the infrastructure configuration or the backends. We have the plan to open source the backends and we have made a public promise to do so, we just do not have the capacity right now to do so.

**2. Accessible - Can community contributors issue pull requests or raise issues on your source code ?**

Yes, our code is hosted on Github (not much yet, but it is slowly growing).

https://github.com/fivebinaries

https://github.com/blockfrost

**3. Transparent - Can community contributors access and propose changes to the source text of your documentation ?**

Yes, the specification of our backends is open source and the community's contributions are welcome (we already got one!) https://github.com/blockfrost/openapi.

**Comment by stephen-rowan** [comment]
Thank you mmahut (Five Binaries), your response has helped a lot.

In response to your identification of the lack of specific criteria to guide responses to the very general question "**Is your Organisation open-source ?**" : 

As you have seen I have only drafted a basic [Basic Open Source Assessment](https://github.com/Quality-Assurance-DAO/DAO-Open-Source/tree/main/Documents/Assessments). This, in part, was deliberate as I want the Catalyst community to help QA-DAO develop the Assessment process itself with us (contribute pull requests and issues) to avoid being prescriptive or adopting an arbitrary authority. 

Naturally this won't be perfect - but I hope we can all learn and refine our processes together.

Standard Response text follows - 

I have captured your responses in the body of the text at [ https://github.com//issues/51 ] and moved the status of the Issue to "Done".

**Note** : The Issue can still be altered and commented upon until Fund 5 Proposals are finalized.

As I set out in the "Detailed explanation" section of each Issue : I will collect all data in a report of findings published on GitHub with a Creative Commons License after Fund 5 Proposals are finalized.

Good Luck with your proposal.

Stephen Whitenstall QA-DAO
